decidim-proposals/app/controllers/decidim/proposals/proposals_controller.rb

```diff
@@ -36,6 +37,28 @@ def create
           end
         end
       end
+
+      private
+
+      def search_klass
+        ProposalSearch
+      end
+
+      def default_search_params
+        {
+          page: params[:page],
+          per_page: 12
+        }
+      end
+
+      def default_filter_params
+        {
+          search_text: "",
+          origin: "all",
+          category_id: "",
+          random_seed: params[:random_seed] || @random_seed
```

Comment:
??? You're passing @random_seed to `search` but @random_seed it's a value from `search`.